packages/medusa/src/repositories/price-list.ts

```diff
@@ -1,39 +1,60 @@
-import { FindOptionsWhere, ILike } from "typeorm"
+import { FindOperator, FindOptionsWhere, ILike, In } from "typeorm"
 import { PriceList } from "../models"
-import { CustomFindOptions, ExtendedFindConfig } from "../types/common"
+import { ExtendedFindConfig } from "../types/common"
 import { dataSource } from "../loaders/database"
 
-export type PriceListFindOptions = CustomFindOptions<
-  PriceList,
-  "status" | "type"
->
-
 export const PriceListRepository = dataSource.getRepository(PriceList).extend({
   async listAndCount(
     query: ExtendedFindConfig<PriceList>,
     q?: string
   ): Promise<[PriceList[], number]> {
     const query_ = { ...query }
+    query_.relationLoadStrategy = "query"
+    query_.where = query.where as FindOptionsWhere<PriceList>
 
-    if (q) {
-      query_.where = query_.where as FindOptionsWhere<PriceList>
-      delete query_.where.description
-      delete query_.where.name
+    const groups = query_.where.customer_groups as unknown as FindOperator<
+      string[]
+    >
+    delete query_.where.customer_groups
+
+    if (groups) {
+      query_.relations = query.relations ?? {}
+      query_.relations.customer_groups =
+        query_.relations.customer_groups ?? true
 
-      query_.where.description = ILike(`%${q}%`)
-      query_.where.name = ILike(`%${q}%`)
       query_.where.customer_groups = {
-        name: ILike(`%${q}%`),
+        ...(query_.where.customer_groups ?? {}),
+        id: In(groups.value),
+      }
+    }
+
+    if (q) {
+      if (!groups) {
+        query.relations = query.relations ?? {}
+        query.relations.customer_groups =
+          query.relations.customer_groups ?? true
       }
+
+      query_.where = [
+        {
+          ...query_.where,
+          name: ILike(`%${q}%`),
+        },
+        {
+          ...query_.where,
+          description: ILike(`%${q}%`),
+        },
+        {
+          ...query_.where,
+          customer_groups: {
+            ...query_.where.customer_groups,
```

Comment:
Q: In line 18, you're deleting customer_groups, why isn't this failing for cases where `groups is falsey`?

Reply:
just lucky I suppose 😂 , I can see some other thing to fix as well

Reply:
Much better. This opens up a new question. :D